sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -0,0 +1,505 @@
+ * @param buffer Can either be an empty #az_span or an #az_span containing properly formatted
+ *               properties starting with the '?' character.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result
+az_iot_hub_client_properties_init(az_iot_hub_client_properties* properties, az_span buffer);
+
+/**
+ * @brief Appends a key-value property to the list of properties.
+ *
+ * @param properties
+ * @param name The name of the property.
+ * @param value The value of the property.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_properties_append(
+    az_iot_hub_client_properties* properties,
+    az_span name,
+    az_span value);
+
+/**
+ * @brief Finds the value of a property.
+ * @note This will return the first value of the property with the given name if multiple properties
+ *       with the same key exist.
+ *
+ * @param properties
+ * @param name The name of the property.
+ * @param out_value An #az_span containing the value of the property.
+ * @return #az_result.
+ */
+AZ_NODISCARD az_result az_iot_hub_client_properties_find(
+    az_iot_hub_client_properties* properties,
+    az_span name,
+    az_span* out_value);
+
+/**
+ * @brief Iterates over the list of properties.
+ *
+ * @param properties
+ * @param out An #az_pair containing the key and the value of the next property.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result
+az_iot_hub_client_properties_next(az_iot_hub_client_properties* properties, az_pair* out);
+
+/**
+ *
+ * Telemetry APIs
+ *
+ */
+
+/**
+ * @brief Gets the MQTT topic that must be used for device to cloud telemetry messages.
+ * @note Telemetry MQTT Publish messages must have QoS At Least Once (1).
+ *
+ * @param client
+ * @param properties An optional properties object (can be NULL).
+ * @param mqtt_topic An empty #az_span with sufficient capacity to hold the MQTT topic.
+ * @param out_mqtt_topic The output #az_span containing the MQTT topic.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_telemetry_publish_topic_get(
+    az_iot_hub_client const* client,
+    az_iot_hub_client_properties const* properties,
+    az_span mqtt_topic,
+    az_span* out_mqtt_topic);
+
+/**
+ *
+ * Cloud to device (C2D) APIs
+ *
+ */
+
+/**
+ * @brief Gets the MQTT topic filter to subscribe to cloud to device requests.
+ * @note C2D MQTT Publish messages will have QoS At Least Once (1).
+ *
+ * @param client
+ * @param mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
+ *                          filter.
+ * @param out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_c2d_subscribe_topic_filter_get(
+    az_iot_hub_client const* client,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief The Cloud To Device Request.
+ *
+ */
+typedef struct az_iot_hub_client_c2d_request
+{
+  az_iot_hub_client_properties

```

Comment:
>az_iot_hub_client_properties [](start = 2, length = 28)

Nit: properties variable is on a different line.

Reply:
Yes: that's the auto-clang format. It's because of the same line doc (related to above comment). I don't like it either.

Reply:
Fixed now using /**<